File: todo.frontend/src/types/todo.ts
export interface Todo {
    id: number;
    title:string;
    description?:string;
    isDone:boolean;
}

export interface CreateTodoDto{
    title:string;
    description?:string;
}

export interface TodoListProps{
    todos:Todo[];
    onToggle: (todoId:number) => void;
}

export interface TodoItemProps{
    todo:Todo;
    onToggle: (todoId:number) => void;
}

export interface TodoFormProps{
    onSubmit: (todoData:CreateTodoDto) => Promise<Todo>;
}

File: Todos.Backend/Todos.Domain/Todo.cs
using System.ComponentModel.DataAnnotations;

namespace Todos.Domain
{
    public class Todo
    {
        [StringLength(100)]
        public string Title { get; set; }

        [StringLength(500)]
        public string Description { get; set; } = string.Empty;

        public int Id { get; set; }
        public bool IsDone { get; set; }
    }
}
